docs/feature_rgblight.md

```diff
@@ -30,6 +30,27 @@ At minimum you must define the data pin your LED strip is connected to, and the
 
 Then you should be able to use the keycodes below to change the RGB lighting to your liking.
 
+### Support on STM32
+
+STM32 chips support driving WS2812B LEDs using their SPI bus. You may want to do this to support both RGB Matrix and Underglow independently, at the same time. Driving via SPI also frees up 2 timers that are used in the alternative implemenetaion.
+
+If you want to use this feature, you have to turn it on specifically, as well as use specific pins and provide some extra information. SPI driven WS2812B LEDs require use of an SPI MOSI pin on your STM32 chip. 
+
+To enable the feature, at a minimum, you must define:
+
+|Define                |Description                                                                |
+|----------------------|---------------------------------------------------------------------------|
+|`RGB_DI_PIN`          |The pin connected to the data pin of the LEDs - this must be a SPI MOSI pin|
+|`RGBLED_NUM`          |The number of LEDs connected                                               |
+|`RGBLIGHT_STM32_SPI`  |Enables the SPI DMA based RGB underglow feature                            |
```

Comment:
I would prefer that this was hardware agnostic.  Eg just:
```suggestion
 |`RGBLIGHT_SPI`  |Enables the SPI DMA based RGB underglow feature                            |
```

Reply:
Wouldnt that confuse people as this library only runs for stm32?

Reply:
Well, does it only run under STM32? Would it work on a Teensy 3.2, for instance?  Which is ARM, but not an STM chip.

Reply:
I haven't tested it on Teensy and the SPI configuration/divisor isn't built out for it. So at this point, only STM32 is supported.

I also don't have a teensy to try to develop it on. In the future it could run, but at the moment, it would require extra code changes. I think we can update the documentation when the code changes happen, but am willing to make the change to docs now if we decide that's what's best.

Reply:
the problem is not the concept of the library , ie using spi as a driver for the leds. Its more rather how the pins/ spi driver is setup for an stm device. Same way we have an i2c library for avr, stm32, magic avr arm chip etc.